### DISCUSSION TITLE
How can I populate the field after get the entity with dataloader?

### DISCUSSION THREAD

**Discussion by wenerme** [mikro-orm/mikro-orm#5443] [General]
I have config `dataloader: DataloaderType.ALL`

```ts
@Resolver((of) => HouseObject)
@Injectable()
export class HouseResolver extends BaseResolverOf({
  ObjectType: HouseObject,
  EntityType: HouseEntity,
  ServiceType: HouseService,
}) {
  @FieldResolver(() => RentContractObject, { nullable: true })
  async lastRentContract(@Root() root: HouseEntity) {

    // works but hav n+1 problem
    await root.populate(['lastRentContract']);

    // not work, nothing happened
    // await root.toReference().load({
    //   populate: ['lastRentContract'],
    //   dataloader: true
    // })
    return root.lastRentContract;
  }
}
```

**Comment by B4nan** [comment]
As said in the other thread, dataloaders are no implemented for `em.populate`, its just for the `Reference/Collection.load`.

So something like this should work:

```ts
async lastRentContract(@Root() root: HouseEntity) {
  return root.lastRentContract?.load();
}
```

(you need to use the `Reference` wrapper in the entity definition, if that's not clear - your example looks more like you are missing that bit too)

**Reply by wenerme** [reply]
@B4nan  Still not works, `root.lastRentContract` is undefined 

no matter `ref: true` or `ref: false`

```ts
  @OneToOne(() => LastRentContractEntity, { mappedBy: (v) => v.house , ref: true})
  lastRentContract?: Rel<LastRentContractEntity>;
```

the inverse side

```ts
export class LastRentContractEntity extends RentContractEntity {
  @OneToOne(() => HouseEntity, { inversedBy: (v) => v.lastRentContract })
  house!: Rel<HouseEntity>;
}
```

**Reply by B4nan** [reply]
Well, the value should be there, and it would be there if this was a regular entity.

**Reply by wenerme** [reply]
But `await root.populate(['lastRentContract']);` this works, because root is inverse side instead of owning side ?

**Reply by B4nan** [reply]
No, because its not a real 1:1 relation, the owning side is virtual entity, right? Check the other thread, this is not supported, you need the `formula` approach.
